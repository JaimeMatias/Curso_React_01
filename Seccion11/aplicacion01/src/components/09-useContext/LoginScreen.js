import React, { useContext } from 'react'

import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { user, setuser } = useContext(UserContext);
  console.log(user)
  const nombre = {
    nombre: "Matias",
    correo: "matias@gmial",
    id: 123
  }
  const handleClick = () => {
    setuser(nombre)
  }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className='btn btn-primary' onClick={handleClick}>LogIn</button>
    </div>
  )
}

