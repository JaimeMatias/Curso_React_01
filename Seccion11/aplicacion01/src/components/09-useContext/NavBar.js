import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-light bg-light">
            <NavLink exact="true"className="navbar-brand" to="/">UseContext</NavLink>

            <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink exact="true"  className="nav-link" aria-current="page" to="/">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact="true"  className="nav-link" to="/about">About</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact="true" className="nav-link" to="/login">Login</NavLink>
                    </li>

                </ul>
            </div>
        </nav>

    )
}
