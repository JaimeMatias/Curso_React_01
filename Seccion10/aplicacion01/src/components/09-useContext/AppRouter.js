import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Link,
    Redirect
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';


export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Routes>

                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route exact path="/" element={<HomeScreen />} /> {/*La ruta mas general debe estar siempre abajo*/}
                    <Route path="*" element={<HomeScreen />} /> {/*Cualquier ruta que no coincida con las anteirores me la panda al HomeScreens*/}

                </Routes>

            </div>
        </Router>

    )
}


