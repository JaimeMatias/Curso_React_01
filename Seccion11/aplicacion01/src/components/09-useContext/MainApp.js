import React, { useState } from 'react'
import { AboutScreen } from './AboutScreen'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'


export const MainApp = () => {
const [user,setuser]=useState({})
  
  return (
    <div>
      <UserContext.Provider value={{user,setuser}}>

        <AppRouter />
      </UserContext.Provider>


    </div>
  )
}

