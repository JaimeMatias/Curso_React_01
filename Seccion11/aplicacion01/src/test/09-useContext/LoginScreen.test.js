import { mount, shallow } from 'enzyme'
import React from 'react'
import { LoginScreen } from '../../components/09-useContext/LoginScreen'
import { UserContext } from '../../components/09-useContext/UserContext'

describe('Pruebas en LoginScreen', () => {
    const user = {
        name: 'Matias1',
        email: 'matias1@gmail.com'
    }
    const setuser = jest.fn()
    const wrapper = mount(
        <UserContext.Provider value={{ user, setuser }}>
            <LoginScreen />
        </UserContext.Provider>
    )
    test('Debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

test('Debe de ejecutar el setUser con el argumento esperado ', () => {
  
    wrapper.find('button').prop('onClick')();
    expect(setuser).toHaveBeenCalledWith({  nombre: "Matias",
    correo: "matias@gmial",
    id: 123});
})

})
