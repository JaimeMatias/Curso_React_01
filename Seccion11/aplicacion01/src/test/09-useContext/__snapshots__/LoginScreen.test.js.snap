// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    LogIn
  </button>
</div>
`;

exports[`Pruebas en LoginScreen Prueba del <LoginScreen>  1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    LogIn
  </button>
</div>
`;
